java/src/main/java/com/google/appengine/tools/mapreduce/MapReduceSettings.java

```diff
@@ -108,17 +110,12 @@ public class MapReduceSettings implements GcpCredentialOptions, ShardedJobAbstra
 
   /**
    * The GCS bucket that will be used for temporary files.
+   *
+   * now REQUIRED to be filled by settings; will not use a default
```

Comment:
I've put some validation when building this, to ensure that is not null, not blanks, whitespaced, etc.

Reply:
it can't be `null`; lombok builder will check that.

yeah - could be whitespace/etc ... it's non-trivial to do validation in lombok builders ...